### PR TITLE
Split auto_expand_albums into three options. Fixes #258

### LINKED ISSUE #258: Don't expand artists when following
I'd like the feature of cmus following the currently played track but because I'm playing random all my artists are opened. Instead I would love cmus to show the currently played track in the artist overview except when the albums of the artist are already showing.

### PR DESCRIPTION
**Pull request by mist**
You can now differ between what to do when searching, following and selecting the currently played track, each of them has their own config options:
auto_expand_albums_search, auto_expand_albums_follow, and auto_expand_albums_selcur.

This fixes #258

### CODE REVIEW

File: tree.c [hunk 1]
```diff
@@ -1249,7 +1249,12 @@ void tree_sort_artists(void)
 
 void tree_sel_current(void)
 {
-	tree_sel_track(lib_cur_track);
+	tree_sel_track(lib_cur_track, auto_expand_albums_selcur);
+}
+
+void tree_sel_current_follow(void)
```
Comment (flyingmutant) [inline tree.c]: Hm, why is this still a separate function, with others like it gone?

**Comment by mist** [comment]
In the first version there were three `tree_sel_track_*` functions which now are gone. The `tree_sel_current` function was called when changing tracks and when pressing i, now it ist only called when pressing i.

**Comment by flyingmutant** [comment]
Now there is also `void tree_sel_current_follow(void)` — I was talking about it.

**Comment by flyingmutant** [comment]
Any updates here?

**Comment by mist** [comment]
I will remove it and change the function to `void tree_sel_current( int auto_expand_albums )`. I've been busy, will do it later this week I think

**Comment by mist** [comment]
Sorry for not hearing from me for so long.

Now it should be as you suggested.

**Comment by flyingmutant** [comment]
Thanks! Merged.

**Pull request merged**